src/main/java/games/strategy/triplea/settings/ClientSetting.java

```diff
@@ -0,0 +1,144 @@
+package games.strategy.triplea.settings;
+
+import java.io.File;
+import java.util.prefs.BackingStoreException;
+import java.util.prefs.Preferences;
+
+import javax.swing.UIManager;
+
+import com.google.common.base.Strings;
+
+import games.strategy.debug.ClientLogger;
+import games.strategy.engine.ClientFileSystemHelper;
+
+/**
+ * List of settings that can be adjusted and stored with a Client's OS. On windows this would be the registry,
+ * these values survive game re-installs. These values can be made available for edit by adding a corresponding
+ * UI binding in {@code ClientSettingUiBinding}. Not all system settings will have UI bindings.
+ *
+ * <p>
+ * Note: After saving values, `ClientSetting.flush()` needs to be called to persist those values.
+ * </p>
+ *
+ *
+ * <p>
+ * Typical usage:
+ * </p><code><pre>
+ *
+ * // loading a value
+ * String value = ClientSetting.AI_PAUSE_DURATION.read();
+ *
+ * // saving value
+ * ClientSetting.AI_PAUSE_DURATION.save(500);
+ * ClientSetting.flush();
+ * </pre></code>
+ */
+public enum ClientSetting {
```

Comment:
I'm not sure if this even works, but this might make stuff easier:
If the enums stored a function object along with the default value which builds the corresponding object from a string value, `#value()` could always return the right object type of the setting, the same could be done reversed for storing values